temporal_sqlalchemy/session.py

```diff
@@ -27,7 +31,54 @@ def persist_history(session: orm.Session, flush_context, instances):
         obj.temporal_options.record_history(obj, session, correlate_timestamp)
 
 
-def temporal_session(session: typing.Union[orm.Session, orm.sessionmaker], strict_mode=False) -> orm.Session:
+def persist_history_on_commit(session: orm.Session, flush_context, instances):
+    if any(_temporal_models(session.deleted)):
+        raise ValueError("Cannot delete temporal objects.")
+
+    metadata = get_session_metadata(session)
+
+    correlate_timestamp = datetime.datetime.now(tz=datetime.timezone.utc)
+    for obj in _temporal_models(itertools.chain(session.dirty, session.new)):
+        if obj.temporal_options.allow_persist_on_commit:
+            metadata[CHANGESET_KEY].add(obj)
+        else:
+            obj.temporal_options.record_history(obj, session, correlate_timestamp)
+
+    if metadata[IS_COMMITTING_KEY]:
+        for obj in metadata[CHANGESET_KEY]:
+            obj.temporal_options.record_history(obj, session, correlate_timestamp)
+
+        metadata[CHANGESET_KEY].clear()
+        metadata[IS_COMMITTING_KEY] = False
+
+
+def enable_is_committing_flag(session):
+    metadata = get_session_metadata(session)
+    metadata[IS_COMMITTING_KEY] = True
+
+
+PERSIST_ON_COMMIT_LISTENERS = (
+    ('before_flush', persist_history_on_commit),
+    ('before_commit', enable_is_committing_flag),
+)
+
+
+PERSIST_LISTENERS = (
+    ('before_flush', persist_history),
+)
+
+
+def set_listeners(session, to_enable=(), to_disable=()):
+    for listener_args in to_disable:
+        if event.contains(session, *listener_args):
+            event.remove(session, *listener_args)
+
+    for listener_args in to_enable:
+        event.listen(session, *listener_args)
+
+
+def temporal_session(session: typing.Union[orm.Session, orm.sessionmaker],
```

Comment:
`temporal_session()` happens during the request setup, right? So it will be a bit tricky to enable the flag on particular endpoints? Could the endpoint instead modify a session which is already been started?

Reply:
It's all dependent on the `persist_on_commit` boolean that is stored in the session's info.  You could manually change it, but I made some modifications to the way `temporal_session` works (not pushed yet) so it can be called multiple times and will change this flag.  This is the same behavior as when you want to turn `strict_mode` on.